inlong-sort/sort-core/src/main/java/org/apache/inlong/sort/parser/impl/FlinkSqlParser.java

```diff
@@ -159,6 +162,23 @@ private void parseStream(StreamInfo streamInfo) {
         log.info("parse stream success, streamId:{}", streamInfo.getStreamId());
     }
 
+    /**
+     * Inject the `inlong.metric` for ExtractNode or LoadNode
+     *
+     * @param streamInfo The encapsulation of nodes and node relations
+     */
+    private void injectInLongMetric(StreamInfo streamInfo) {
+        streamInfo.getNodes().stream().filter(node -> node instanceof InLongMetric).forEach(node -> {
+            Map<String, String> properties = node.getProperties();
+            if (properties == null) {
+                properties = new LinkedHashMap<>();
+            }
```

Comment:
if properties is null, new LinkedHashMap should be writed back to node by node.setProperties

Reply:
done